### PR TITLE
Gaussian Blur concept

### PR DESCRIPTION
**Pull request by spaceribs**
This isn't in a great place yet, but I'm submitting it for feedback. BlurX and BlurY deprecated in favor of BlurDir, which will allow some nice motion blurs later. This should also be a partial fix (and inspired by) for #1652 but it's still:
1. A little too griddy.
2. Strangely still based on the viewport.
Anyway, I'm putting this in for more eyes and feedback.

### CODE REVIEW

**Comment by spaceribs** [comment]
Leaving these 2 shadertoy filters here for some reference material:
* https://www.shadertoy.com/view/XdfGDH
* https://www.shadertoy.com/view/Mtl3Rj

File: src/filters/blur/BlurDirFilter.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+BlurDirFilter.prototype.constructor = BlurDirFilter;
+module.exports = BlurDirFilter;
+
+BlurDirFilter.prototype.applyFilter = function (renderer, input, output, clear) {
+
+    var shader = this.getShader(renderer);
+
+    this.uniforms.strength.value = this.strength / 4 / this.passes * (input.frame.width / input.size.width);
+
+    if (this.passes === 1) {
```
Comment (englercj) [inline src/filters/blur/BlurDirFilter.js]: Please fix the braces in this block to use the newline-first style used throughout the code base :)

File: src/filters/blur/BlurDirFilter.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+BlurDirFilter.prototype = Object.create(core.AbstractFilter.prototype);
+BlurDirFilter.prototype.constructor = BlurDirFilter;
+module.exports = BlurDirFilter;
+
+BlurDirFilter.prototype.applyFilter = function (renderer, input, output, clear) {
+
+    var shader = this.getShader(renderer);
+
+    this.uniforms.strength.value = this.strength / 4 / this.passes * (input.frame.width / input.size.width);
```
Comment (englercj) [inline src/filters/blur/BlurDirFilter.js]: Magic number `4` here, prefer a constant that describes what it is for.

File: src/filters/blur/BlurFilter.js [hunk 3]
```diff
@@ -26,12 +35,14 @@ BlurFilter.prototype.applyFilter = function (renderer, input, output)
 {
     var renderTarget = renderer.filterManager.getRenderTarget(true);
 
-    this.blurXFilter.applyFilter(renderer, input, renderTarget);
-    this.blurYFilter.applyFilter(renderer, renderTarget, output);
+    for (var e = 0; e < this.blurFilters.length; e++) {
```
Comment (englercj) [inline src/filters/blur/BlurFilter.js]: `e` seems like a random iterator, why not `i`?

**Comment by englercj** [comment]
Looks good to me, had some minor nitpicks about style I mentioned. Also, wouldn't this supersede the BlurX/Y filters? We should remove them if so.

**Comment by spaceribs** [comment]
Hey engle, thanks for taking a look!

Yeah I was definitely thinking about just taking them out entirely, the other option is hooking them up to 2 BlurDirs in opposite directions if you wanted to keep that.

I'll fix those style issues today, but I also kinda want to implement one of these high quality blurs as well (linked in the shadertoys). Let me know if you think we should make it an entirely separate filter.

**Comment by GoodBoyDigital** [comment]
Nice work @spaceribs!How does this perform compared to our current implementation?

**Comment by spaceribs** [comment]
I can set that up, do you guys have something for benchmarking I can use?

**Comment by englercj** [comment]
@spaceribs No, we don't have a framework setup for benchmarking specifically.

**Comment by englercj** [comment]
This looks good to me, if you remove the old filters that this replaces I will merge it.

**Comment by GoodBoyDigital** [comment]
I would like to keep both for now peeps! At least until we have done some perf tests. ta!

**Comment by englercj** [comment]
Done!

**Comment by spaceribs** [comment]
Hey sorry, I've been working on other stuff all day, last weekend I also
worked on a separate fine blur based on the shaders linked above, I'll open
up another PR for it though

**Pull request merged**